src/PrestaShopBundle/Routing/LegacyUrlConverter.php

```diff
@@ -0,0 +1,268 @@
+    }
+
+    /**
+     * @param array $parameters
+     * @param string $routeName
+     *
+     * @return array
+     */
+    private function convertParametersForRoute(array $parameters, $routeName)
+    {
+        $route = $this->legacyRoutes[$routeName];
+        $routeDefaults = $route->getDefaults();
+        if (!empty($routeDefaults['_legacy_parameters']) && is_array($routeDefaults['_legacy_parameters'])) {
+            foreach ($routeDefaults['_legacy_parameters'] as $legacyParameter => $parameter) {
+                if (isset($parameters[$legacyParameter])) {
+                    $parameters[$parameter] = $parameters[$legacyParameter];
+                    unset($parameters[$legacyParameter]);
+                }
+            }
+        }
+
+        $legacyAction = $this->getActionFromParameters($parameters);
+        unset($parameters['controller']);
+        unset($parameters['action']);
+        unset($parameters[$legacyAction]);
+
+        return $parameters;
+    }
+
+    /**
+     * @param array $parameters
+     *
+     * @return string
+     *
+     * @throws RouteNotFoundException
+     */
+    private function findRouteNameByParameters(array $parameters)
+    {
+        $legacyController = $parameters['controller'];
+        $controllerActions = $this->controllersActions[$legacyController];
+
+        $legacyAction = $this->getActionFromParameters($parameters);
+        if (!isset($controllerActions[$legacyAction])) {
+            throw new RouteNotFoundException(
+                'Could not find a route matching for legacy action: %s',
+                [$legacyController . ':' . $legacyAction]
+            );
+        }
+
+        return $controllerActions[$legacyAction];
+    }
+
+    /**
+     * @param array $parameters
+     *
+     * @return null|string
+     */
+    private function getActionFromParameters(array $parameters)
+    {
+        $legacyAction = null;
+
+        if (!empty($parameters['action'])) {
+            $legacyAction = $parameters['action'];
+        } else {
+            //Actions can be defined as simple query parameter (e.g: ?controller=AdminProduct&add)
+            $controllerActions = array_keys($this->controllersActions[$parameters['controller']]);
+            foreach ($parameters as $parameter => $value) {
+                if (in_array($parameter, $controllerActions)) {
+                    $legacyAction = $parameter;
+                    break;
+                }
+            }
+        }
+
+        return $this->isIndexAction($legacyAction) ? 'index' : $legacyAction;
+    }
+
+    /**
+     * @param Route $route
+     *
+     * @return array
+     */
+    private function breakLegacyLink(Route $route)
+    {
+        $routeDefaults = $route->getDefaults();
+        $legacyLink = $routeDefaults['_legacy_link'];
+        $linkParts = explode(':', $legacyLink);
+        $legacyController = $linkParts[0];
+        $legacyAction = count($linkParts) > 1 ? $linkParts[1] : null;
```

Comment:
`isset($linkParts[1])` is maybe faster than count?

Reply:
good question, I can change it anyway and even use empty to be sure
But regarding the performance.. do you know which one is faster?